worker/server_state.go

```diff
@@ -118,19 +118,25 @@ func (s *ServerState) initStorage() {
 
 	{
 		// Write Ahead Log directory
-		x.Checkf(os.MkdirAll(Config.WALDir, 0700), "Error while creating WAL dir.")
+		if Config.WALDir != "" {
```

Comment:
@harshil-goel just trying to understand the changes. 

Is it the case that when badger is run in-memory, WALDir and PostingDir will be empty? And raftwal.InitEncrypted() handles this case properly?